src/CommunityToolkit.Aspire.Hosting.Python.Extensions/StreamlitAppHostingExtension.cs

```diff
@@ -0,0 +1,104 @@
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Python;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Streamlit applications to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class StreamlitAppHostingExtension
+{
+    /// <summary>
+    /// Adds a Streamlit application to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the Streamlit application.</param>
+    /// <param name="appDirectory">The path to the directory containing the Streamlit application.</param>
+    /// <param name="scriptPath">The path to the Python script to be run by Streamlit (relative to appDirectory).</param>
+    /// <returns>An <see cref="IResourceBuilder{StreamlitAppResource}"/> for the Streamlit application resource.</returns>
+    /// <remarks>
+    /// <para>
+    /// This method uses the Aspire.Hosting.Python integration to run Streamlit applications.
+    /// By default, it uses the <c>.venv</c> virtual environment in the app directory.
+    /// Use standard Python extension methods like <c>WithVirtualEnvironment</c>, <c>WithPip</c>, or <c>WithUv</c> to customize the environment.
+    /// </para>
+    /// <para>
+    /// **⚠️ EXPERIMENTAL:** This integration is experimental and subject to change. The underlying implementation
+    /// will be updated to use public APIs when they become available in Aspire.Hosting.Python (expected in Aspire 13.1).
+    /// </para>
+    /// </remarks>
+    /// <example>
+    /// Add a Streamlit application to the application model:
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// builder.AddStreamlitApp("dashboard", "../streamlit-app", "app.py")
+    ///        .WithHttpEndpoint(env: "PORT");
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    [Experimental("CTASPIRE001", UrlFormat = "https://github.com/CommunityToolkit/Aspire/issues/{0}")]
+    public static IResourceBuilder<StreamlitAppResource> AddStreamlitApp(
```

Comment:
Does debugging and deploying work?

Reply:
I think we should make sure it works. The bar should be higher in 13 for new language / framework integrations. Even if it doesn't work I don't want us to merge things without understanding what works and what doesn't work. I'd expect the future to look like a set of capabililities for these language integrations (otel support, endpoints, https, dockerfile, debugging). Doing enough work to know what we will document is a minimum bar here.

Reply:
While I appreciate the desire, we have to work with the realities of the OSS project - we don't have the resources to do everything that Aspire does.

The goal is to do our best to be producing high quality integrations to the level of Aspire, but it's not always going to be the case, especially if we want to ship. Feel free to raise an issue to get debugger support enabled, but I don't see it as a blocker for us shipping.